core/unit_test/TestGraph.hpp

```diff
@@ -1100,4 +1118,125 @@ TEST(TEST_CATEGORY, graph_then) {
   ASSERT_TRUE(contains(exec, data, value_memset + value_then));
 }
 
+template <typename DataViewType, typename BufferViewType>
+struct ThenIncrementAndCombineFunctor
+    : public IncrementAndCombineFunctor<DataViewType, BufferViewType> {
+  using base_t = IncrementAndCombineFunctor<DataViewType, BufferViewType>;
+
+  KOKKOS_FUNCTION void operator()() const { base_t::operator()(0); }
+};
+
+template <typename T>
+struct GraphIsDefaulted : std::true_type {};
+
+#if defined(KOKKOS_ENABLE_CUDA) ||                                           \
+    (defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)) || \
+    (defined(KOKKOS_ENABLE_SYCL) && defined(KOKKOS_IMPL_SYCL_GRAPH_SUPPORT))
+template <>
+struct GraphIsDefaulted<
+    Kokkos::Experimental::Graph<Kokkos::DefaultExecutionSpace>>
+    : std::false_type {};
+#endif
+
+template <typename T>
+constexpr bool is_graph_defaulted = GraphIsDefaulted<T>::value;
```

Comment:
same here ...`is_defaulted_graph` ... but I guess you already follow a convention